src/v/security/acl.cc

```diff
@@ -216,6 +218,31 @@ acl_store::acls(const acl_binding_filter& filter) const {
     return result;
 }
 
+ss::future<fragmented_vector<acl_binding>> acl_store::all_bindings() const {
+    fragmented_vector<acl_binding> result;
+    for (const auto& acl : _acls) {
+        for (const auto& entry : acl.second) {
+            result.push_back(acl_binding{acl.first, entry});
+            co_await ss::coroutine::maybe_yield();
+        }
+    }
+    co_return result;
+}
+
+ss::future<>
+acl_store::reset_bindings(const fragmented_vector<acl_binding>& bindings) {
+    // NOTE: not coroutinized because otherwise clang-14 crashes.
+    _acls.clear();
+    return ss::do_for_each(
+             bindings,
+             [this](const auto& binding) {
+                 _acls[binding.pattern()].insert(binding.entry());
```

Comment:
i guess there isn't any reason to go through the acl_store api rather than directly to the underlying data structure?

Reply:
Not sure why this is needed, but `acl_store::add_bindings` does a rehash after every insert. I guess this is a bad idea for a bulk reset if the number of bindings is substantial.

Reply:
good point.